packages/propeller/src/Tracing.ts

```diff
@@ -6,37 +6,37 @@ export type EventHandler<Format> = (
   events: ITimelineEvent<Format>[],
 ) => Promise<void>
 
-export async function traceHandler<Format>(
+export function traceHandler<Format>(
   tracer: Tracer,
   attrs: Attributes,
-  streamName: StreamName,
-  events: ITimelineEvent<Format>[],
   handler: EventHandler<Format>,
 ) {
-  const category = StreamName.category(streamName)
-  const firstEventTimeStamp = events[events.length - 1]!.time.getTime()
-  return tracer.startActiveSpan(
-    `${category} process`,
-    {
-      kind: SpanKind.CONSUMER,
-      attributes: {
-        [Tags.category]: category,
-        [Tags.stream_name]: streamName,
-        "eqx.stream_version": Number(events[events.length - 1]!.index),
-        [Tags.loaded_count]: events.length,
-        ...attrs,
+  return function tracedHandler(streamName: StreamName, events: ITimelineEvent<Format>[]) {
+    const category = StreamName.category(streamName)
+    const firstEventTimeStamp = events[events.length - 1]!.time.getTime()
+    return tracer.startActiveSpan(
+      `${category} process`,
+      {
+        kind: SpanKind.CONSUMER,
+        attributes: {
+          [Tags.category]: category,
+          [Tags.stream_name]: streamName,
+          "eqx.stream_version": Number(events[events.length - 1]!.index),
+          [Tags.loaded_count]: events.length,
+          ...attrs,
+        },
       },
-    },
-    (span) =>
-      handler(streamName, events)
-        .catch((err) => {
-          span.recordException(err)
-          span.setStatus({ code: SpanStatusCode.ERROR, message: err.message })
-          throw err
-        })
-        .finally(() => {
-          span.setAttribute("eqx.lead_time_ms", Date.now() - firstEventTimeStamp)
-          span.end()
-        }),
-  )
+      (span) =>
+        handler(streamName, events)
+          .catch((err) => {
+            span.recordException(err)
+            span.setStatus({ code: SpanStatusCode.ERROR, message: err.message })
+            throw err
+          })
+          .finally(() => {
+            span.setAttribute("eqx.lead_time_ms", Date.now() - firstEventTimeStamp)
```

Comment:
eqx branding overuse - ideally stuff to do with queues of events should have diff label to aid people splitting stuff in their heads